test/sparse.fieldsets.spec.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+const _ = require('lodash')
+const Promise = require('bluebird')
+const Joi = require('joi')
+const Hapi = require('hapi')
+
+let server, buildServer, destroyServer, hh;
+
+const schema = {
+    type: 'brands',
+    attributes: {
+        code: Joi.string().min(2).max(10),
+        description: Joi.string()
+    }
+};
+
+const data = {
+    attributes: {
+        code: 'MF',
+        description: 'Massey Furgeson'
+    }
+};
+
+
+//TODO just done the validation, actual includes is remaining
+describe('Sparse Fieldsets', function() {
+    
+    beforeEach(function(done) {
+        buildServer(() => {
+            let promises = [];
+        
+            _.times(10, () => {
+                promises.push(server.injectThen({method: 'post', url: '/brands', payload: {data}}))
+            })
+            
+            return Promise.all(promises)
+            .then(() => {
+                done()
+            })
+        })
+    })
+    
+    afterEach(function(done) {
+        destroyServer(done)
+    })
+    
+    it('Will be able to GET all from /brands with a sparse fieldset', function() {
```

Comment:
First 2 tests seem to be false positives
The fields clause is not used in the Mongodb adapter and will not affect the results

See the other comment on mixup of fields and include vocabulary